actors/market/src/lib.rs

```diff
@@ -1025,6 +1008,122 @@ impl Actor {
             }
         }
     }
+
+    fn settle_deal_payments(
+        rt: &impl Runtime,
+        params: SettleDealPaymentsParams,
+    ) -> Result<SettleDealPaymentsReturn, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+        let curr_epoch = rt.curr_epoch();
+
+        let mut batch_gen = BatchReturnGen::new(params.deal_ids.len());
+        let mut settlements: Vec<DealSettlementSummary> = Vec::new();
+        // accumulates slashed amounts from timed out deal proposals that weren't activated in time
+        let mut total_slashed = TokenAmount::zero();
+
+        rt.transaction(|st: &mut State, rt| {
+            let mut new_deal_states: Vec<(DealID, DealState)> = Vec::new();
+            for deal_id in params.deal_ids {
+                let deal_proposal = match st.get_proposal(rt.store(), deal_id) {
+                    Ok(prop) => prop,
+                    Err(_) => {
+                        batch_gen.add_fail(ExitCode::USR_NOT_FOUND); // hide potential EX_DEAL_EXPIRED as deal may have terminated
+                        continue;
+                    }
+                };
+                let dcid = match rt_deal_cid(rt, &deal_proposal) {
+                    Ok(cid) => cid,
+                    Err(e) => {
+                        batch_gen.add_fail(e.exit_code());
+                        continue;
+                    }
+                };
+
+                let loaded_deal = match st.get_active_deal_or_process_timeout(
+                    rt.store(),
+                    curr_epoch,
+                    deal_id,
+                    &deal_proposal,
+                    &dcid,
+                ) {
+                    Ok(res) => res,
+                    Err(e) => {
+                        batch_gen.add_fail(e.exit_code());
+                        continue;
+                    }
+                };
+
+                let mut deal_state = match loaded_deal {
+                    LoadDealState::TooEarly => {
+                        // deal is not active, we process it as a zero-payment no-op
+                        settlements.push(DealSettlementSummary {
+                            completed: false,
+                            payment: TokenAmount::zero(),
+                        });
+                        continue;
+                    }
+                    LoadDealState::ProposalExpired(penalty) => {
+                        // deal proposal was not activated in time
+                        total_slashed += penalty;
+                        batch_gen.add_fail(ExitCode::USR_NOT_FOUND);
```

Comment:
Use `EX_DEAL_EXPIRED` for consistency with other places deals are looked up. I'll update the FIP to match.